File: core/src/ru/myitschool/dcrawler/effects/ControlledEffect.java
package ru.myitschool.dcrawler.effects;

import com.badlogic.gdx.graphics.Texture;
import ru.myitschool.dcrawler.entities.Entity;

/**
 * Created by Voyager on 24.11.2017.
 */
public class ControlledEffect extends Effect {

    private String name = "Mind control";
    private String description = "Someone else controls your mind for this turn.";
    private Texture icon = new Texture("mind_control.png");
    private boolean used = false;
    private boolean positive = false;

    public ControlledEffect(Entity entity) {
        super(entity);
        entity.setControlled(true);
    }

    @Override
    public Texture getIcon() {
        return icon;
    }

    @Override
    public String getName() {
        return name;
    }

    @Override
    public String getDescription() {
        return description;
    }

    @Override
    public boolean isSkillUse() {
        return false;
    }

    @Override
    public boolean isPositive() {
        return false;
    }

    @Override
    public boolean isStackable() {
        return false;
    }

    @Override
    public int getStackSize() {
        return 0;
    }

    @Override
    public boolean isExpiring() {
        return true;
    }

    @Override
    public int getExpireTurns() {
        return 1;
    }

    @Override
    public String getType() {
        return "main.dcrawler.effect.controlled";
    }

    @Override
    public void endTurn() {
        super.endTurn();
        getEntity().setControlled(false);
    }
}

File: core/src/com/wolg_vlad/dcrawler/effects/ControlledEffect.kt
package com.wolg_vlad.dcrawler.effects

import com.badlogic.gdx.graphics.Texture
import com.wolg_vlad.dcrawler.entities.Entity

/**
 * Created by Voyager on 24.11.2017.
 */
class ControlledEffect(entity: Entity?) : Effect(entity) {
    private val used = false
    override val name = "Mind control"
    override val description = "Someone else controls your mind for this turn."
    override val icon = Texture("mind_control.png")
    override val positive: Boolean = false
    override val skillUse: Boolean = false
    override val stackable: Boolean = false
    override val stackSize: Int = 0
    override val expiring: Boolean = true
    override val expireTurns: Int = 1
    override val id: String = "main.dcrawler.effect.controlled"

    override fun endTurn() {
        super.endTurn()
        entity!!.isControlled = false
    }

    init {
        entity!!.isControlled = true
    }
}